Core/HLE/sceCtrl.cpp

```diff
@@ -204,6 +204,12 @@ void __CtrlButtonUp(u32 buttonBit)
 	ctrlCurrent.buttons &= ~buttonBit;
 }
 
+void __CtrlSetAllButtons(u32 buttonBits)
+{
+	std::lock_guard<std::mutex> guard(ctrlMutex);
+	ctrlCurrent.buttons = buttonBits & CTRL_MASK_USER;
```

Comment:
Not everything goes through the control mapper, such as API events and tilt.  I think it'd be much better if this took two parameters, one to clear and one to set.

That way if the API is setting X pressed and the user is not doing anything, it won't incorrectly clear it.  Currently, this would incorrectly clear it AFAICT.

-[Unknown]

Reply:
That's a good point, and will fit well with my WIP code. Will change before merging this.